vendor/bat-native-ledger/src/bat/ledger/internal/publisher/server_publisher_fetcher.cc

```diff
@@ -0,0 +1,317 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/publisher/server_publisher_fetcher.h"
+
+#include <utility>
+
+#include "base/big_endian.h"
+#include "base/json/json_reader.h"
+#include "base/strings/string_piece.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/common/brotli_helpers.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/publisher/channel_response.pb.h"
+#include "bat/ledger/internal/publisher/prefix_util.h"
+#include "bat/ledger/internal/request/request_publisher.h"
+#include "bat/ledger/option_keys.h"
+#include "brave_base/random.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace {
+
+constexpr size_t kQueryPrefixBytes = 2;
+
+int64_t GetCacheExpiryInSeconds(bat_ledger::LedgerImpl* ledger) {
+  DCHECK(ledger);
+  // NOTE: We are reusing the publisher prefix list refresh interval for
+  // determining the cache lifetime of publisher details. At a later
+  // time we may want to introduce an additional option for this value.
+  return ledger->GetUint64Option(ledger::kOptionPublisherListRefreshInterval);
+}
+
+ledger::PublisherStatus PublisherStatusFromMessage(
+    const publishers_pb::ChannelResponse& response) {
+  switch (response.wallet_connected_state()) {
+    case publishers_pb::UPHOLD_ACCOUNT_KYC:
+      return ledger::PublisherStatus::VERIFIED;
+    case publishers_pb::UPHOLD_ACCOUNT_NO_KYC:
+      return ledger::PublisherStatus::CONNECTED;
+    default:
```

Comment:
let's not use default as then we can miss if new ones are added

Reply:
Unfortunately, for protobuf generated code, we get compile errors if we leave off the default.